app/models/nav.rb

```diff
@@ -1,9 +1,53 @@
 class Nav
-  def initialize(nav_data = [])
-    @nav_data = nav_data
+  attr_reader :data, :path
+
+  def initialize(path, data = [])
+    @data = data.freeze
+    @path = path
+  end
+
+  def current_item_root
+    data.find { |item| item["name"] == current_item[:parents][0] }
   end
 
-  def nav_tree
-    @nav_data
+  # Returns the current nav item
+  def current_item
+    item = route_map[path.sub("/docs/", "")]
+    raise ActionController::RoutingError.new("Missing navigation for #{path}") unless item
+    item
+  end
+
+  # Returns a hash of routes, indexed by path
+  #
+  # @example
+  #   {
+  #     "path/to/page": {
+  #       path: "path/to/page",
+  #       parents: ["parent", "grandparent"]
```

Comment:
It's worth calling out that I'm doing route matching by keeping track of ancestors since our URLs aren't representative of our navigation structure. If that were the case then path pattern matching would be much simpler!

e.g.

This is the current routing and the parent that should be highlighted.
`/docs/tutorials` – [Pipelines]

Ideally in this case we'd be using `/docs/pipelines/tutorials/`, for simplicities sake, since we'd then be able to determine Pipelines should be highlighted based purely on the URL. I'm aware there's more to consider before making this change, redirects etc. but if we do ever make that change then we should be able to remove a bunch of this code.

Reply:
👍🏻 Even when we look into updating the URLs, we may not want to include section names. That makes it easier to rename sections, move pages, and reduces the URL depth. So all the pages for pipelines may just be `/pipelines/*`. Something to consider later.